README.md

```diff
@@ -1,2 +1,104 @@
+```
+where `:controllers` contains the `:start` and `:stop` functions that are called when entering and leaving the route, respectively. `:view` is a property unique to re-route that specifies a Reagent render function.
+
+`default-route` is the `:name` of the fallback route to be served instead of a route that is not found in the router. This is usually for "Not Found" pages.
+
+`prevent-nav-opts` is an options map for navigation prevention:
+- `:enabled?` determines if nav prevention is enabled or not. If `false`, then the `::re-route/set-prevent-nav` effect handler has no effect. Default `true`.
+- `:back-button?` determines whether nav prevention is enabled for browser back button clicks (or other changes to the browser history stack). Default `false`, as enabling this causes weird effects with the browser history, due to limitations of the History API.
+- `:default-text` sets the default text for confirm dialogs. By default this is "You have unsaved changes, are you sure?"
+
+#### `[::re-route/navigate route-name path-params query-params fragment]`
+
+Navigate to a new route with these parameters, and calls `.pushState` on the history stack.
+
+#### `[::re-route/navigate-replace route-name path-params query-params fragment]`
+
+Same as `::re-route/navigate`, but calls `.replaceState` on the history stack instead.
+
+#### `[::re-route/set-prevent-nav ?text]`
+
+Sets navigation prevention, which will cause navigation away from the current page to trigger a confirm dialog. `?text` sets the confirm dialog text (for non-`beforeunload` dialogs).
+
+#### `[::re-route/unset-prevent-nav]`
+
+Unsets navigation prevention, allowing navigation to proceed as normal again.
+
+#### `[::re-route/on-start]` and `[::re-route/on-stop]`
+
+Calls the `on-start` and `on-stop` multimethods, respectively.
+
+### Subscriptions
+
+#### `[::re-route/href route-name path-params query-params fragment]`
+
+Creates a link from the parameters to be used in anchor tags. Clicking on the link is the same as navigation via `::re-route/navigate` for the same route and parameters.
+
+#### `[::re-route/prevent-nav]`
+
+Returns the current `prevent-nav` map.
+
+#### `[::re-route/route]`
+
+Returns the current route map.
+
+#### `[::re-route/route-name]`
+
+Returns the name of the current route.
+
+#### `[::re-route/route-view]`
+
+Returns the `:view` render function of the current route.
+
+#### `[::re-route/path-params]`, `[::re-route/query-params]`, `[::re-route/fragment]`
+
+Returns the appropriate parameters of the current route.
+
+## Run Demo
+
+This library comes with a demo that demonstrates the functionality of re-route. This can be run via Figwheel or can be compiled and run on an nginx webserver.
+
+To run via Figwheel, run `make dev`.
+
+To compile the webpage, run `make bundle`, which will create a `target/bundle` directory. Then to run the webpage on an nginx Docker image, `cd` to the `target/bundle` directory, then run:
+```
+docker build -t re-route-nginx-image .
```

Comment:
So I guess we never got figwheel to work on its own?

Reply:
Well nginx was the approach Cliff suggested and it worked. Is figwheel _supposed_ to work on its own?